exercises/bhuvan_bezawada/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+* PROGRAM DESCRIPTION:
+*
+* A command line program to sort shapes and display the area and perimeter 
+* of each shape.
+*
+**/
+
+// Imports
+#include <algorithm>
+#include <iostream>
+#include <math.h>
+#include <string.h>
+#include <string>
+#include <vector>
+
+// Namespace being used
+using namespace std;
+
+/**
+ *  Base class
+ **/
+class Shape {
+
+// Instance variables
+private:
+  string shapeType;
+  unsigned int numSides;
+
+// Methods
+public:
+  Shape(const string &type, int sides) {
+    shapeType = type;
+    numSides = sides;
+  }
+
+  // Methods to be overridden
+  virtual double perimeter() = 0;
+  virtual double area() = 0;
+
+  // Methods for info
+  void printType() const {
```

Comment:
I would not hard-code the IO like this, I would return a string and leave it up to the client code to determine how to best present/persist it.